### ISSUE TITLE
30s Preview

### ISSUE THREAD

**Issue by SiddharthS018** [spotify/web-api#1420]
I have integrated Spotify with a service & when I try to play the track using the song Link, I am able to view only the 30s Preview & It says Listen to the full song on Spotify!, Can anyone please say me why I am not able to play the full song?

**Comment by 0xNF** [comment]
It sounds like you may have used [the play button widget](https://developer.spotify.com/documentation/widgets/guides/adding-a-spotify-play-button/), which has some very limited capabilities. 

If you are not logged in, it will only play the 30 second clip, like you described. To listen to the full track, you need to be logged in to Spotify. The button will then play the full song via either the web player or the installed Spotify client on the user's platform.

Basically, log into to Spotify and you'll get the behavior you want.

**Comment by SiddharthS018** [comment]
Hi,
I didn't use the play button widget, instead, I tried to render the music
Play.
I have signed in to Spotify & I have a Premium account.

Shall I know what causes it to play a random 30s from the song?

**Comment by 0xNF** [comment]
Hm, then I'm not sure what you've done. Could you explain in more detail how exactly you integrated and "rendered" the music?

**Comment by SiddharthS018** [comment]
Hello,

The integration is with Zoho Cliq,
The integrations are done using oAuth2 with cliq.

Here I want to play the music in the chat window of a Bot.
All the links posted in bot will be unfurled & when I post the link of the
song , I get the link Rendered in a card as the image attached with this
mail.
The query is that , I want to play the full music in the bot, but I am able
to view only a 30s preview.
But other than Spotify, I was able to get a full duration play from
(Youtube, soundcloud, etc).

Please help out getting a full duration play , instead of a 30s preview.

Best,
Siddharth S

**Comment by SiddharthS018** [comment]
Hi,
I am waiting for your support?.

**Comment by kevinguebert** [comment]
It depends on the Spotify service you are using.
The Spotify Web API only returns a 30s preview clip called the `preview_url`. [See the documentation for a Track as reference.](https://developer.spotify.com/documentation/web-api/reference/tracks/get-track/)
```
preview_url | string | A link to a 30 second preview (MP3 format) of the track. Can be null
-- | -- | --
```

However, you could look into implementing another service from Spotify that does support playback. 
If you are using a browser, it sounds like the [Web Playback SDK](https://developer.spotify.com/documentation/web-playback-sdk/) would probably be your best bet.

**Comment by SiddharthS018** [comment]
No, the issue I am facing is ,
If I post the url of youtube / SoundCloud , a card will be rendered & there
will be full-time play inside the bot.(Note that there is no integration
between youtube/soundCloud with Zoho Cliq).
But in case of Spotify , there is only an 30s preview if the link is posted
in the bot.
Are you providing only 30s preview, can you extend it to a fullplay?

Pleases check the screenshots attached below.
1.Spotify 30s playback
2.Youtube full playback

Note:
'Using Refurling Method there is a full play of Youtube video in the bot &
only the link is posted with no integrations.
Please try to find the issue in refurling in Spotify to provide a complete
playback'

**Comment by SiddharthS018** [comment]
Hi,
If you are unable to identify the issue, Please say me.
Only if you respond, I can proceed further with development.

**Comment by 0xNF** [comment]
I'd like to point out that both Kevin and myself are in no way affiliated with Spotify and aren't obligated or expected to provide support. We are just fellow users of the code in this repo.

It's hard to answer your questions because your attached screenshots got stripped, and aren't appearing in this thread, so we can't see exactly what you are experiencing. It should be noted that the _only_ way to get Spotify to play songs from an unofficial client (like your bot) is via the player API, which I do not think you are using. 

Youtube and Soundcloud have more lenient playback policies, which is why you are getting the full tracks via those other services.

**Comment by SiddharthS018** [comment]
Sorry, I sent the reply from mail & didn't check here, 
I will attach Here.
1. [Spotify 30s playback](https://drive.google.com/open?id=1qAnlrG6TRMX0bnEiAHc0tGmGA-s_DfvD)
2. [Youtube full playback](https://drive.google.com/open?id=12AGjnq19lTd2AUvGbYp3vmEKIp9c_qMk) 

Please note that only the link is posted in the bot in both the cases.
The card is rendured using Refurling Method.

**Comment by SiddharthS018** [comment]
If I can't get the exact answer here please help me out, where I can request for support @0xNF @kevinguebert, As I am a Premium User I guess I would get good support.

**Comment by kevinguebert** [comment]
It looks like this ticket and #1430 are the very similar. Both have widgets, both only playing 30 seconds

**Comment by SiddharthS018** [comment]
In Slack, if we post a link to a bot, a card is rendered & complete music play takes place inside .. 
But in case of Zoho cliq , it's not so. 
Only the 30s of play is available. 

What will be the reason because both are using refuelling method , but only 30s play in Zoho cliq. 

Here I am not talking about APIs. Only refurling method !

**Comment by 0xNF** [comment]
Sounds like Spotify might be doing some behind the scenes detection for whether it plays full or preview versions of the songs. Maybe they have a partnership with slack that lets them do so. But if it gets routed through Zoho beforehand, it doesn't work. Just speculating here.

**Comment by SiddharthS018** [comment]
The Issue is fixed, & it isn't from the Spotify side, 
Thank you @0xNF ,@kevinguebert  for your support.
